### PR TITLE
feat: 슬랙 로그인 리프레시 토큰 구현

### PR DESCRIPTION
**Pull request by Daco2020**
또봇 서버에 리프레시 토큰 로직을 구현합니다.

### CODE REVIEW

**Comment by Daco2020** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/Daco2020/ttobot/48?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#48** <a href="https://app.graphite.dev/github/pr/Daco2020/ttobot/48?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#47** <a href="https://app.graphite.dev/github/pr/Daco2020/ttobot/47?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @Daco2020 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: app/api/auth.py [hunk 1]
```diff
@@ -82,19 +82,24 @@ async def current_user(
...
             status_code=403, detail=f"토큰이 유효하지 않습니다. token: {token}"
         )
 
-    user_id = result.get("user_id", "")
+    if decoded_payload.get("type") == "refresh":
+        raise HTTPException(
+            status_code=403, detail=f"토큰이 유효하지 않습니다. token: {token}"
+        )
+
+    user_id = decoded_payload.get("user_id", "")
```
Comment (Daco2020) [inline app/api/auth.py:96]: 리프레시 토큰으로는 로그인이 되지 않도록 합니다.

File: app/api/views/login.py [hunk 2]
```diff
@@ -53,11 +56,48 @@ async def slack_auth(
...
+
+@router.get("/slack/auth/refresh")
+async def slack_auth_refresh(
+    refresh_token: str,
+    service: ApiService = Depends(api_service),
+):
+    try:
+        decoded_payload = decode_token(refresh_token)
+        if decoded_payload.get("type") != "refresh":
+            return HTTPException(status_code=403, detail="토큰이 유효하지 않습니다.")
```
Comment (Daco2020) [inline app/api/views/login.py:84]: 리프레시 토큰만 액세스 토큰을 재발급 받을 수 있도록 검증합니다.

**Review by nayoung0 — APPROVED** [review]
오예 리프레시 토큰 감사합니다 ‼️

**Pull request merged**